x/mongo/driver/topology/errors.go

```diff
@@ -4,8 +4,12 @@ import (
 	"fmt"
 
 	"go.mongodb.org/mongo-driver/mongo/description"
+	"go.mongodb.org/mongo-driver/x/mongo/driver"
 )
 
+// Assert that ConnectionError satisfies driver.RetryablePoolError.
+var _ driver.RetryablePoolError = ConnectionError{}
```

Comment:
The Retryable Writes spec says:

> When the driver encounters a network error establishing an initial connection to a server,
  it MUST add a RetryableWriteError label to that error if the MongoClient performing
  the operation has the retryWrites configuration option set to true.

IIUC, the `RetryableWriteError` is already applied to network errors that occur during the handshake commands.
The [handshaker](https://github.com/mongodb/mongo-go-driver/blob/9101b419216336cec091fc70d1f622efa12f5ca0/mongo/client.go#L435-L439) will eventually call `operation.Execute`.

Does this need to be applied to other network errors? For example, if [configureTLS](https://github.com/mongodb/mongo-go-driver/blob/9101b419216336cec091fc70d1f622efa12f5ca0/x/mongo/driver/topology/connection.go#L197-L200) returns an error?